packages/api/src/modules/user/services/stats/user.stats.service.ts

```diff
@@ -57,6 +58,33 @@ export class UserStatsService {
             await notifyService.notify(NotificationType.USER_ALREADY_EXIST, data);
         }
     }
+
+    async updateAllUsersInSubTools() {
+        const users = await userRepository.findAll();
+
+        for (const user of users) {
+            if (user.disable) continue;
+
+            let reset: null | UserReset = null;
+            if (!user.active) {
+                reset = await userResetRepository.findOneByUserId(user._id);
+            }
+
+            const data = {
+                email: user.email,
+                firstname: user.firstName,
+                lastname: user.lastName,
+                agentType: user.agentType,
+                jobType: user.jobType,
+                url: reset ? `${FRONT_OFFICE_URL}/auth/reset-password/${reset.token}?active=true` : undefined,
+                active: user.active,
+                signupAt: user.signupAt,
+                lastActivityDate: user.lastActivityDate,
+            } as NotificationDataTypes[NotificationType.USER_UPDATED];
+
+            await notifyService.notify(NotificationType.USER_UPDATED, data);
```

Comment:
pas mieux de await un `Promise.all(notifyService.notify[])` ?

Reply:
Il n'y a qu'un notify..? Je ne comprends pas ta proposition

Reply:
`await` dans le `for.. of` à la place d'un seul `await Promise.all()` à la fin de la boucle

Reply:
Ah oui en effet. Par ailleurs je suis mécontente de ce code parce qu'il nous expose à des [429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) (brevo limite à 10 requêtes par seconde et j'ai l'impression qu'on pourrait être + rapide que ça, surtout si on fait un `Promise.all`.
Je suis tentée d'utiliser le point d'api d'[update multiple](https://developers.brevo.com/reference/updatebatchcontacts) mais ça me paraît overkill de faire un `NotificationType` pour cet usage uniquement qui est très localisé donc je suis tentée de faire l'appel brevo directement :point_right: :point_left: 
Je pense que si je fais ça c'est @VSerain qui râlerait

Reply:
Les deux me vont de mon côté !

Reply:
oh ! bonne nouvelle pour moi

Reply:
Il me semble qu'on avait déjà fait à un endroit un système de pile qu'on ferait coter pipe, ça ne changerait rien d'un point de vue perf, ça éviterait juste qu'on ai une logique spécifique à brevo dans le user stat, mais je pense que c'est pas forcément le lieux de ce fix

Reply:
système de pile qui attend + Promise.await , c'est ma tournée ^^